controllers/namespace_controller.go

```diff
@@ -101,11 +101,13 @@ func (r *NamespaceReconciler) propagateMeta(ctx context.Context, ns, parent *cor
 		}
 	}
 
-	subNS := &accuratev1.SubNamespace{}
-	err := r.Get(ctx, types.NamespacedName{Name: ns.Name, Namespace: parent.Name}, subNS)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("failed to get sub namespace %s/%s: %w", ns.Name, parent.Name, err)
-	} else {
+	if _, ok := ns.Labels[constants.LabelParent]; ok {
+		subNS := &accuratev1.SubNamespace{}
+		err := r.Get(ctx, types.NamespacedName{Name: ns.Name, Namespace: parent.Name}, subNS)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get sub namespace %s/%s: %w", ns.Name, parent.Name, err)
+		}
```

Comment:
What happens if `apierrors.IsNotFound(err)` is `true`?
In that case, subNS is invalid.  Although the following code would work for such invalid data,
I want to check them and skip code using `subNS`.